lib/ansible/modules/database/postgresql/postgresql_publication.py

```diff
@@ -638,7 +638,7 @@ def main():
 
     # Get final publication info:
     pub_fin_info = {}
-    if state != 'absent' or (state == 'absent' and module.check_mode):
+    if state == 'present' or (state == 'absent' and module.check_mode):
```

Comment:
Only states are present & absent, so this doesn't change anything, though this PR makes it clearer.